test/no-await-in-promise-methods.mjs

```diff
@@ -0,0 +1,187 @@
+import outdent from 'outdent';
+import {getTester} from './utils/test.mjs';
+
+const {test} = getTester(import.meta);
+
+const error = {
+	messageId: 'no-await-in-promise-methods',
+};
+
+test({
+	valid: [
+		outdent`
+			Promise.all([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise[all]([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise.allSettled([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise[allSettled]([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise.any([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise[any]([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise.race([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise[race]([promise, promise, promise, promise])
+		`,
+		outdent`
+			Promise.resolve([await promise])
+		`,
+	],
+
+	invalid: [
+		{
+			code: outdent`
+				Promise.all([promise, await promise, await promise, promise])
+			`,
+			errors: [error, error],
+			output: outdent`
+				Promise.all([promise, promise, promise, promise])
+			`,
+		},
+		{
+			code: outdent`
+				Promise[all]([promise, await promise, await promise, promise])
+			`,
+			errors: [error, error],
+			output: outdent`
+				Promise[all]([promise, promise, promise, promise])
+			`,
+		},
+		{
+			code: outdent`
+				Promise.all([await (promise, promise)])
+			`,
+			errors: [error],
+			output: outdent`
+				Promise.all([(promise, promise)])
+			`,
+		},
+		{
+			code: outdent`
+				Promise.all([, await promise])
+			`,
+			errors: [error],
+			output: outdent`
+				Promise.all([, promise])
+			`,
+		},
+		{
+			code: outdent`
+				Promise.allSettled([promise, await promise, await promise, promise])
+			`,
+			errors: [error, error],
+			output: outdent`
+				Promise.allSettled([promise, promise, promise, promise])
+			`,
+		},
+		{
+			code: outdent`
+				Promise[allSettled]([promise, await promise, await promise, promise])
+			`,
+			errors: [error, error],
+			output: outdent`
+				Promise[allSettled]([promise, promise, promise, promise])
+			`,
+		},
+		{
+			code: outdent`
+				Promise.allSettled([await (promise, promise)])
+			`,
+			errors: [error],
+			output: outdent`
+				Promise.allSettled([(promise, promise)])
+			`,
+		},
+		{
+			code: outdent`
+				Promise.allSettled([, await promise])
+			`,
+			errors: [error],
+			output: outdent`
+				Promise.allSettled([, promise])
+			`,
+		},
+		{
+			code: outdent`
+				Promise.any([promise, await promise, await promise, promise])
+			`,
+			errors: [error, error],
+			output: outdent`
+				Promise.any([promise, promise, promise, promise])
```

Comment:
This changes logic, can't fix.

Reply:
I turned the auto-fix into a suggestion.